solver/src/driver.rs

```diff
@@ -197,7 +201,51 @@ impl Driver {
             self.min_order_age,
             &mut settlements,
         );
-        let settlements = self.simulate_settlements(settlements).await?;
+
+        // Remove liquidity provision interactions from settlements.
```

Comment:
Instead of adding all this code here could we maybe do the extra simulation just before we submit the settlement?


```rust
if let Some(settlement) = settlements
  .into_iter()
  .max_by(|a, b| a.objective_value.cmp(&b.objective_value))
{
  // If we have enough buffer in the settlement contract to not use on-chain interactions, remove those
  if can_settle(settlement.without_liquidity()) {
    settlement = settlement.without_liquidity()
  }
  self.submit_settlement(settlement).await;
}
```


I think it depends a little bit how phase 1 and 2 would be implemented. Would the logic of which/how to replace interactions go in the SettlementEncoder or would this logic here stay around?

Reply:
Right, I was getting ahead of myself since on the second phase we won't need the extra simulation, and will do everything before the final simulation. But I agree it makes more sense that we don't make it unnecessarily complex in this PR. Done in c9118ec.